src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/IInterceptingPropertyValueProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// A project property provider that intercepts all the callbacks for a specific property name
+    /// on the default <see cref="IProjectPropertiesProvider"/> for validation and/or transformation of the property value.
+    /// </summary>
+    internal interface IInterceptingPropertyValueProvider
+    {
+        /// <summary>
+        /// Gets the name of the property.
+        /// </summary>
+        string GetPropertyName();
```

Comment:
This should be metadata value not a method, so that you don't instantiate these things unless actually needed.